src/gmp_agents.c

```diff
@@ -314,8 +316,6 @@ modify_agents_run (gmp_parser_t *gmp_parser, GError **error)
 
       if (uuid && is_uuid (uuid))
         g_ptr_array_add (uuid_array, g_strdup (uuid));
```

Comment:
I think there should be an `else` case returning an error and also an error if the sub-element does not have the tag `agent`.